istioctl/cmd/istioctl/dashboard.go

```diff
@@ -0,0 +1,501 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"runtime"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/istioctl/pkg/kubernetes"
+
+	// import all known client auth plugins
+	v1 "k8s.io/api/core/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	dashboardCmd = &cobra.Command{
+		Use:     "dashboard",
```

Comment:
pass multiple dashboard types?

Reply:
could we also have an all value that just opens up every available dashboard?

Reply:
We have five "global" dashboard types: grafana, kiali, prometheus, jaeger, and zipkin.  A user can't install both zipkin and jaeger.  Except during a demo I probably know which dashboard type I need.

We have two "pod" dashboards.  envoy (only pods with sidecars) and controlz (a few control pane pods) that could not be part of "all".

Reply:
We could maybe use the word available? We should be able to work that out using Kube API server calls.

Reply:
Let's get some experience with single types.  I would like to do a follow-on later for integrating the various 3rd party dashboards.